lib/prompts.js

```diff
@@ -0,0 +1,45 @@
+"use strict";
+
+var _ = require("lodash");
+var async = require("async");
+var inquirer = require("inquirer");
+
+/**
+ * Prompt user for input, validate, and add derived fields to final data object.
+ *
+ * @param {Object}    init      Initialization configuration (`prompts`, `derived`)
+ * @param {Function}  callback  Calls back with `(err, data)`
+ * @returns {void}
+ */
+module.exports = function (init, callback) {
+  if (!init) { return callback(new Error("Invalid init object")); }
+
+  // Mutate `prompts` into appropriate format.
+  var prompts = init.prompts || [];
+  if (!_.isArray(prompts)) {
+    if (!_.isObject(prompts)) {
+      return callback(new Error("Invalid prompts type: " + typeof prompts));
+    }
+
+    prompts = _.map(prompts, function (val, key) {
+      return _.extend({ name: key }, val);
+    });
+  }
+
+  // Execute prompts, then derive final data.
+  async.auto({
+    prompts: function (cb) {
+      // Get user prompts. No error, because will prompt user for new input.
+      inquirer.prompt(prompts, function (data) { cb(null, data); });
+    },
+    derived: ["prompts", function (cb, results) {
+      // Create object of functions bound to user input data and invoke.
+      async.auto(_.mapValues(init.derived, function (transform) {
```

Comment:
Why is `async.auto` used here when there's only one function call that is not async?